listenbrainz/listenstore/influx_listenstore.py

```diff
@@ -384,20 +386,40 @@ def dump_listens(self, location, dump_time=datetime.today(), threads=None):
                 # get listens from all measurements and write them to files in
                 # a temporary dir before adding them to the archive
                 for user in users:
-                    query = 'SELECT * FROM ' + get_escaped_measurement_name(user['musicbrainz_id'])
+                    username = user['musicbrainz_id']
+                    offset = 0
+                    listens = []
 
-                    # loop until we get this user's listens
+                    # Get this user's listens in chunks
                     while True:
-                        try:
-                            results = self.influx.query(query)
+
+                        # loop until we get this chunk of listens
+                        while True:
+                            try:
+                                result = self.influx.query("""
```

Comment:
Given that this dump is about to listens up to, but not beyond the dump timestamp, this query should be limited to all listens up to the timestamp.